File: front-end/src/application/repository/accessRepository.ts
import { Access } from "../../domain/entity/access";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";

export interface AccessRepository {
    save: (access: Access) => Promise<Either<DataError, Access>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Access[]>>;
    
    deleteAll: () => Promise<Either<DataError, Access[]>>;
}

File: back-end/src/main/java/com/trucks/application/port/AccessRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import com.trucks.domain.entity.Access;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

public interface AccessRepository {

    Access save(Access access);

    Page<Access> findAll(PageRequest page);

    Access update(UUID id, Access access);

    List<Access> deleteAll();
}
